src/app/pages/StoryPage/StoryPage.jsx

```diff
@@ -131,6 +139,30 @@ const StoryPage = ({ pageData, mostReadEndpointOverride }) => {
     group5: 4,
   };
 
+  const { enabled: adsEnabled } = useToggle('ads');
```

Comment:
Is this check needed? the Ad container already does this for us

Reply:
It's needed to render the `CanonicalAdBootstrap` as there's no such check in that container

Reply:
Hmmmm okay, thanks. Seems odd that we check the same toggle twice. Perhaps this can be refactored at a later date

Reply:
Could we create a refactor issue for this @rhenshaw56 ? I'm wondering if we just keep the check in the `StoryPage` and remove it from the pre-proc rules. That way the check is only done once? I wouldn't stop this PR though.

Reply:
@andrewcfc removed in [this commit](https://github.com/bbc/simorgh/pull/7582/commits/1b690f820705112817c16c09a004b303cfcfbec0) as it's a one liner

Reply:
In my opinion, we shouldn't be adding an MPU block to the page data if allowAdvertising is false. This is why I implemented this check. Pre-processing isn't free and if we don't need to deep clone large objects into memory for transformation on every getInitialData call, then we shouldn't. The check in the StoryPage is for the leaderboard, which is not managed by the pre-processor.

Reply:
Fair enough @simonsinclair, I guess this is code readability vs. performance and performance takes precedence here. Leave the issue or any further changes then imo 👍